### PR TITLE
Prepare templates before release

### PR DESCRIPTION
**Pull request by pawelprazak**
- add a GHA workflow
- add smoke tests using pulumi preview
- temporarily revert syntax to work with the released 0.0.1-beta version

Depends on #176

### CODE REVIEW

File: .github/workflows/templates.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+on:
+  push:
+    branches: [ 'main', 'develop' ]
+    paths:
+      - 'template'
```
Comment (prolativ) [inline .github/workflows/templates.yml:5]:
```suggestion
      - 'templates'
```

File: .github/workflows/templates.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  pull_request:
+    branches: [ 'main', 'develop' ]
+    paths:
+      - 'template'
```
Comment (prolativ) [inline .github/workflows/templates.yml:9]:
```suggestion
      - 'templates'
```

File: .github/workflows/templates.yml (hunk 1 above)
Comment (lbialy) [inline .github/workflows/templates.yml:5]: templates

File: .github/workflows/templates.yml (hunk 2 above)
Comment (lbialy) [inline .github/workflows/templates.yml:9]: templates

File: templates/kubernetes/Main.scala [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-import besom.api.kubernetes.apps.v1.{Deployment, DeploymentArgs}
+import besom.api.kubernetes.apps.v1.{deployment, DeploymentArgs}
 import besom.api.kubernetes.core.v1.inputs.{ContainerArgs, ContainerPortArgs, PodSpecArgs, PodTemplateSpecArgs}
 import besom.api.kubernetes.meta.v1.inputs.{LabelSelectorArgs, ObjectMetaArgs}
 import besom.api.kubernetes.apps.v1.inputs.DeploymentSpecArgs
 
 @main def main = Pulumi.run {
   val appLabels = Map("app" -> "nginx")
-  for nginxDeployment <- Deployment(
+  for nginxDeployment <- deployment(
```
Comment (lbialy) [inline templates/kubernetes/Main.scala:9]: wait, what?

**Review by lbialy — CHANGES_REQUESTED** [review]
we need this to be updated to 0.1.0 once we go live

File: templates/kubernetes/Main.scala (hunk 3 above)
Comment (pawelprazak) [reply templates/kubernetes/Main.scala:9]: beta thawa was released has old syntax as mentioned on slack, will update to 0.2 in next step

File: .github/workflows/templates.yml (hunk 2 above)
Comment (pawelprazak) [reply .github/workflows/templates.yml:9]: thank you! :)

**Comment by pawelprazak** [comment]
> we need this to be updated to 0.1.0 once we go live

yes, I agree, but I'd prefer to merge this and do 0.1.0 update in next PR

**Comment by pawelprazak** [comment]
Merged #176 here to resolve cyclic dependency

**Review by lbialy — APPROVED** [review]

**Pull request merged**